### PR TITLE
Adds version status endpoint. Factors workflow status from VersionService

### LINKED ISSUE #4779: Provide new DSA versions endpoint for version state
It should return the current version, whether the item is open, openable, assembling, accessioning, and closable.

### PR DESCRIPTION
**Pull request by justinlittman**
closes #4779

## Why was this change made? 🤔
To centralize versioning logic.

## How was this change tested? 🤨

⚡ ⚠ If this change has cross service impact, including data writes to shared file systems, ***run [integration tests](https://github.com/sul-dlss/infrastructure-integration-test)*** and/or test in [stage|qa] environment, in addition to specs. ⚡

Unit

### CODE REVIEW

File: app/controllers/objects_controller.rb [hunk 1]
```diff
@@ -83,7 +83,7 @@ def accession
     EventFactory.create(druid: params[:id], event_type: 'accession_request', data: { workflow: })
 
     # if this object is currently already in accessioning, we cannot start it again
-    if VersionService.in_accessioning?(druid: @cocina_object.externalIdentifier, version: @cocina_object.version)
+    if WorkflowStateService.accessioning?(druid: @cocina_object.externalIdentifier, version: @cocina_object.version)
```
Comment (justinlittman) [inline app/controllers/objects_controller.rb:86]: Versioning related workflow logic is extracted to WorkflowStateService.

File: app/controllers/versions_controller.rb [hunk 2]
```diff
@@ -49,6 +49,20 @@ def openable
     render build_error('Unable to check if openable due to preservation client error', e, status: :internal_server_error)
   end
 
+  def status
```
Comment (justinlittman) [inline app/controllers/versions_controller.rb:52]: New endpoint for getting version status info.

File: app/services/version_service.rb [hunk 3]
```diff
@@ -105,9 +107,7 @@ def can_open?(assume_accessioned: false)
   def close(description:, user_name:, event_factory:, start_accession: true)
     ObjectVersion.update_current_version(druid:, description:) if description
 
-    raise VersionService::VersioningError, "Trying to close version #{version} on #{druid} which is not opened for versioning" unless open_for_versioning?
-    raise VersionService::VersioningError, "Trying to close version #{version} on #{druid} which has active assemblyWF" if active_assembly_wf?
-    raise VersionService::VersioningError, "accessionWF already created for versioned object #{druid}" if accessioning?
+    ensure_closeable!
```
Comment (justinlittman) [inline app/services/version_service.rb:110]: Abstracted into a separate method to support new `can_close?` method.

File: app/services/workflow_state_service.rb [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+  def initialize(druid:, version:)
+    @druid = druid
+    @version = version
+  end
+
+  ASSEMBLY_WORKFLOWS = %w[assemblyWF etdAssemblyWF gisAssemblyWF wasCrawlPreassemblyWF wasSeedPreassemblyWF].freeze
+
+  # Checks if the latest version has any assembly workflows with incomplete steps.
+  # @return [Boolean] true if object is currently being assembled
+  def assembling?
```
Comment (justinlittman) [inline app/services/workflow_state_service.rb:19]: This is the only new method. The rest are extracted from VersionService.

File: app/services/workflow_state_service.rb [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+    ASSEMBLY_WORKFLOWS.any? { |workflow| active_workflow?(workflow:) }
+  end
+
+  # The following methods were extracted from VersionService.
+  # As such, they may not represent the current best practice for determining workflow state
+  # and will probably be subject to further refactoring or removal.
```
Comment (mjgiarlo) [inline app/services/workflow_state_service.rb:25]: 👏🏻

**Review by mjgiarlo — APPROVED** [review]

File: app/services/workflow_state_service.rb [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+  def self.accessioning?(...)
+    new(...).accessioning?
+  end
+
+  def initialize(druid:, version:)
+    @druid = druid
+    @version = version
+  end
+
+  ASSEMBLY_WORKFLOWS = %w[assemblyWF etdAssemblyWF gisAssemblyWF wasCrawlPreassemblyWF wasSeedPreassemblyWF].freeze
```
Comment (mjgiarlo) [inline app/services/workflow_state_service.rb:15]: @justinlittman Assuming we're not renaming the workflow, I believe the ETD-related workflow is `etdSubmitWF` not `etdAssemblyWF` ❓

**Pull request merged**